tests/ludwig/data/test_split.py

```diff
@@ -262,3 +262,59 @@ def random_date(*args, **kwargs):
 
         assert np.all(split["date_col"] > min_datestr)
         min_datestr = split["date_col"].max()
+
+
+@pytest.mark.parametrize(
+    ("df_engine",),
+    [
+        pytest.param(PandasEngine(), id="pandas"),
+        pytest.param(DaskEngine(_use_ray=False), id="dask", marks=pytest.mark.distributed),
+    ],
+)
+def test_hash_split(df_engine, ray_cluster_2cpu):
+    nrows = 100
+    npartitions = 10
+
+    df = pd.DataFrame(np.random.randint(0, 100, size=(nrows, 3)), columns=["A", "B", "C"])
+    df["id"] = np.arange(0, 100)
+
+    if isinstance(df_engine, DaskEngine):
+        df = df_engine.df_lib.from_pandas(df, npartitions=npartitions)
+
+    probabilities = [0.8, 0.1, 0.1]
+    split_params = {"type": "hash", "column": "id", "probabilities": probabilities}
+    splitter = get_splitter(**split_params)
+
+    backend = Mock()
+    backend.df_engine = df_engine
+    splits = splitter.split(df, backend)
+
+    assert len(splits) == 3
+
+    for split, p in zip(splits, probabilities):
+        if isinstance(df_engine, DaskEngine):
+            split = split.compute()
+
+        # Should be approximately the same size as the desired proportion
+        assert nrows * p - 5 <= len(split["id"]) <= nrows * p + 5
+
```

Comment:
Check IDs are disjoint across splits

Needs `from itertools import combinations`
```suggestion

    # IDs should not overlap between splits
    assert all([set(split1["id"]).isdisjoint(set(split2["id"])) for split1, split2 in combinations(splits, 2)])

```